### ISSUE TITLE
Improve Search Results

### ISSUE THREAD

**Issue by jkodroff** [pulumi/registry#558]
The search feature on the registry doesn't work the way I would expect. I searched for "azure", but I would've expected the first result to be the Azure Native and Classic providers, but I got a bunch of lower-level Azure docs. Maybe this could work similar to how Searchlight reports categorized results, e.g. searching for "azure":

* Providers:
    * Azure Native
    * Azure Classic
* (other categories of content TBD - I don't know the registry well enough to suggest something)

**Comment by susanev** [comment]
@kimberleyamackenzie i think we may be able to tie this into the "improve docs user experience" initiative for this quarter. i believe @cnunciato has given you admin access to swifttype. let me know if you have interest in contributing to making this better.

**Comment by kimberleyamackenzie** [comment]
@susanev yeah, for sure! definitely makes sense as part of that initiative + a great chance to get more familiar with swifttype generally for me

**Comment by cnunciato** [comment]
Thanks for the help on this! Hit me up if you have any questions as you get going.

**Comment by kimberleyamackenzie** [comment]
I have weighted the tags as described above.  A couple speedbumps we're running into:

- Using Azure as an example of something we have native and classic versions of:  with the weighted tags along, searching "azure" still didn't bring the package landing pages to the top.  This is because the package titles and overview language are actually "Azure Native" and "Azure Classic."  I solved this for our featured packages by customizing the search results and pinning the landing pages for classic + native to the top of the results.
- But, this is related to a larger challenge when titles on the page don't necessarily match the search terms folks might be using.  A good example of this is https://www.pulumi.com/registry/packages/azuread/.  If I search "AzureAD", with the new weighting, the landing page is the first result.  However, if I search "Azure active directory," the overview doesn't make it into the first page of results, because we don't use that full name in the tags or title.  We can work around this for one-off instances by setting synonyms in swiftype - but we should consider doing a pass to make sure we have all of these set where we want them to be, or otherwise update the meta descriptions to use full names.

I'm going to call this issue complete, and I have created https://github.com/pulumi/registry/issues/762 to dive deeper into the full name search issue.
